x-pack/plugins/observability_solution/inventory/public/hooks/use_discover_redirect.ts

```diff
@@ -7,23 +7,22 @@
 import { useCallback, useMemo } from 'react';
 import type { InventoryEntity } from '../../common/entities';
 import { useAdHocDataView } from './use_adhoc_data_view';
-import { useFetchEntityDefinition } from './use_fetch_entity_definition';
+import { useFetchEntityDefinitionIndexPattern } from './use_fetch_entity_definition_index_patterns';
 import { useKibana } from './use_kibana';
 
 export const useDiscoverRedirect = (entity: InventoryEntity) => {
   const {
     services: { share, application, entityManager },
   } = useKibana();
-  const { entityDefinitions, isEntityDefinitionLoading } = useFetchEntityDefinition(
-    entity.entityDefinitionId
-  );
+  const { entityDefinitionIndexPatterns, isEntityDefinitionIndexPatternsLoading } =
+    useFetchEntityDefinitionIndexPattern(entity.entityType);
```

Comment:
Entities of the same type are expected to have the same index pattern, right?


Calling `useFetchEntityDefinitionIndexPattern` in `use_discover_redirect` means that for each row a request to `inventory/entity/definitions/sources?type={type}` will be triggered. 

What do you think of sending the request only when the panel is expanded? Alternatively, we could fetch the existing entity types, such as services and hosts, in a single request:

and request `inventory/entity/definitions/sources?types=services,hosts`

This could return a response like:


```
definitionIndexPatterns: {
   "services": "metrics-apm*, logs-apm*",
    "hosts": "metrics-infra*" 
}

```

This would reduce redundant requests and improve scalability. What do you think?